apps/studio/components/interfaces/Organization/BillingSettings/CreditTopUp.tsx

```diff
@@ -0,0 +1,369 @@
+import { AlertCircle, Info } from 'lucide-react'
+import { useFlag } from 'hooks/ui/useFlag'
+import { ButtonTooltip } from 'components/ui/ButtonTooltip'
+import { toast } from 'sonner'
+import { useQueryClient } from '@tanstack/react-query'
+import { subscriptionKeys } from 'data/subscriptions/keys'
+import HCaptcha from '@hcaptcha/react-hcaptcha'
+import { useIsHCaptchaLoaded } from 'stores/hcaptcha-loaded-store'
+
+const stripePromise = loadStripe(STRIPE_PUBLIC_KEY)
+
+const FORM_ID = 'credit-top-up'
+
+const FormSchema = z.object({
+  amount: z.coerce
+    .number()
+    .gte(100, 'Amount must be between $100 - $999.')
+    .lte(999, 'Amount must be between $100 - $999.')
+    .int('Amount must be a whole number.'),
+  paymentMethod: z.string(),
+})
+
+type CreditTopUpForm = z.infer<typeof FormSchema>
+
+export const CreditTopUp = ({ slug }: { slug: string | undefined }) => {
+  const { resolvedTheme } = useTheme()
+  const queryClient = useQueryClient()
+
+  const creditTopUpEnabled = useFlag('creditTopUp')
+  const canTopUpCredits = useCheckPermissions(
+    PermissionAction.BILLING_WRITE,
+    'stripe.subscriptions'
+  )
+  const isPermissionsLoaded = usePermissionsLoaded()
+
+  const {
+    mutate: topUpCredits,
+    isLoading: executingTopUp,
+    error: errorInitiatingTopUp,
+  } = useOrganizationCreditTopUpMutation({})
+
+  const form = useForm<CreditTopUpForm>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      amount: 100,
+      paymentMethod: '',
+    },
+  })
+
+  const [topUpModalVisible, setTopUpModalVisible] = useState(false)
+  const [paymentConfirmationLoading, setPaymentConfirmationLoading] = useState(false)
+  const captchaLoaded = useIsHCaptchaLoaded()
+  const [captchaToken, setCaptchaToken] = useState<string | null>(null)
+  const [captchaRef, setCaptchaRef] = useState<HCaptcha | null>(null)
+
+  const captchaRefCallback = useCallback((node: any) => {
+    setCaptchaRef(node)
+  }, [])
+
+  const resetCaptcha = () => {
+    setCaptchaToken(null)
+    captchaRef?.resetCaptcha()
+  }
+
+  const initHcaptcha = async () => {
+    if (topUpModalVisible && captchaRef && captchaLoaded) {
+      let token = captchaToken
+
+      try {
+        if (!token) {
+          const captchaResponse = await captchaRef.execute({ async: true })
+          token = captchaResponse?.response ?? null
+          setCaptchaToken(token)
+          return token
+        }
+      } catch (error) {
+        return token
+      }
+
+      return token
+    }
+  }
+
+  useEffect(() => {
+    initHcaptcha()
+  }, [topUpModalVisible, captchaRef, captchaLoaded])
+
+  const [paymentIntentSecret, setPaymentIntentSecret] = useState('')
+  const [paymentIntentConfirmation, setPaymentIntentConfirmation] = useState<PaymentIntentResult>()
+
+  const onSubmit: SubmitHandler<CreditTopUpForm> = async ({ amount, paymentMethod }) => {
+    setPaymentIntentConfirmation(undefined)
+
+    const token = await initHcaptcha()
+
+    await topUpCredits(
```

Comment:
no need for await here since this is `mutate` not `mutateAsync`

Reply:
I think it's still cleaner to use await given this is an async function that can fail and it's clearer this way, especially if someone adds a line after

Reply:
hmm if thats the case then swap the above to use `mutateAsync`? otherwise the await isn't waiting here